src/updater/updater.cpp

```diff
@@ -149,446 +95,568 @@ void CheckWeightsConsistency(
+
+            turn_duration_penalties[edge_index] = turn_duration_penalty;
+            turn_weight_penalties[edge_index] = turn_weight_penalty;
+            updated_turns.push_back(edge_index);
         }
 
-        for (std::size_t i = 0; i < merged_counters.size(); i++)
+        if (turn_weight_penalty < 0)
         {
-            if (i == LUA_SOURCE)
-            {
-                util::Log() << "Used " << merged_counters[LUA_SOURCE]
-                            << " speeds from LUA profile or input map";
-            }
-            else
-            {
-                // segments_speeds_counters has 0 as LUA, segment_speed_filenames not, thus we need
-                // to susbstract 1 to avoid off-by-one error
-                util::Log() << "Used " << merged_counters[i] << " speeds from "
-                            << config.segment_speed_lookup_paths[i - 1];
-            }
+            util::Log(logWARNING) << "Negative turn penalty at " << turn_index.from_id << ", "
+                                  << turn_index.via_id << ", " << turn_index.to_id
+                                  << ": turn penalty " << turn_weight_penalty;
         }
     }
 
-    const auto maybe_save_geometries = [&] {
-        if (!update_edge_weights)
-            return;
+    return updated_turns;
+}
+}
+
+Updater::NumNodesAndEdges Updater::LoadAndUpdateEdgeExpandedGraph() const
+{
+    std::vector<extractor::EdgeBasedEdge> edge_based_edge_list;
+    std::vector<EdgeWeight> node_weights;
+    auto max_edge_id = Updater::LoadAndUpdateEdgeExpandedGraph(edge_based_edge_list, node_weights);
+    return std::make_tuple(max_edge_id + 1, std::move(edge_based_edge_list));
+}
 
-        // Now save out the updated compressed geometries
-        extractor::io::write(config.geometry_path, segment_data);
-    };
+EdgeID
+Updater::LoadAndUpdateEdgeExpandedGraph(std::vector<extractor::EdgeBasedEdge> &edge_based_edge_list,
+                                        std::vector<EdgeWeight> &node_weights) const
+{
+    TIMER_START(load_edges);
 
-    const auto save_datastore_names = [&] {
-        extractor::Datasources sources;
-        DatasourceID source = 0;
-        sources.SetSourceName(source++, "lua profile");
+    auto max_edge_id = 0;
 
-        // Only write the filename, without path or extension.
-        // This prevents information leakage, and keeps names short
-        // for rendering in the debug tiles.
-        for (auto const &name : config.segment_speed_lookup_paths)
-        {
-            sources.SetSourceName(source++, boost::filesystem::path(name).stem().string());
-        }
+    {
+        storage::io::FileReader reader(config.edge_based_graph_path,
+                                       storage::io::FileReader::VerifyFingerprint);
+        auto num_edges = reader.ReadElementCount64();
+        edge_based_edge_list.resize(num_edges);
+        max_edge_id = reader.ReadOne<EdgeID>();
```

Comment:
Huh interesting that this is not an error / warning at all - you're reading a u32 into a i32 here.